### PR TITLE
Index for lockarray should be non-negative

### PR DESCRIPTION
**Pull request by reddycharan**
Descriptions of the changes in this PR:

Make sure lockIndex is non-negative.

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ BufferedLogChannel getLogChannel() {
         private BufferedLogChannelWithDirInfo entryLogWithDirInfo;
 
         private EntryLogAndLockTuple(long ledgerId) {
-            int lockIndex = Long.hashCode(ledgerId) % lockArrayPool.length();
+            int lockIndex = Math.abs(Long.hashCode(ledgerId) % lockArrayPool.length());
```
Comment (merlimat) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java]: If `hashCode()` returns a negative, the modulo operation will fail anyway. I think we should use `MathUtils.signSafeMod()` here.

Comment (reddycharan) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java]: https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.17.3

in java for modulo (remainder) operation both right and left operands can be negative.

**Comment by reddycharan** [comment]
@sijie @eolivelli fyi this is critical fix

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java (hunk 1 above)
Comment (merlimat) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java]: Sure, though I'd prefer to consistently use `signSafeMod()` like in rest of codebase

Comment (reddycharan) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java]: changed it to signSafeMod

**Pull request closed without merging**